src/schema.ts

```diff
@@ -182,17 +199,80 @@ export class Schema {
 }
 
 /**
- * @return a summary with the correct distinct property given a max number of bins
+ * @return a summary of the binning scheme determined from the given max number of bins
  */
-function binSummary(maxbins: number, summary: Summary) {
-  const bin = dlBin({
+function binSummary(maxbins: number, summary: Summary): Summary {
+  const bin: Bin = dl.bins({
     min: summary.min,
     max: summary.max,
     maxbins: maxbins
   });
-  const binSum = extend({}, summary);
-  binSum.distinct = (bin.stop - bin.start) / bin.step;
-  return binSum;
+
+  return binStats(bin, summary);
+}
+
+function timeUnitSummary(unit: TimeUnit, summary: Summary, type: Type): Summary {
+  var bin: Bin;
+  switch (unit) {
+    // these are the units that dl.bins.date supports
```

Comment:
Why do you think it doesn't work?  It's just called differently.  
(If it really doesn't works and is buggy, file a bug in datalib.)

> Because the dl.bins.date method doesn't work with that unit.

You will need to convert Vega-Lite's multi-unit (e.g., `YEARMONTH`) to DL's corresponding pluralized periodic units (e.g., `months`). 

Some VL time units don't have 1:1 correspondents in DL, so we can ignore those cases for now.